src/web-ui/src/public/Cart.vue

```diff
@@ -144,9 +144,14 @@ export default {
     checkout: function () {
       this.$router.push('/checkout');
     },
-    triggerAbandonedCartEmail : function () {
-      console.log("Xo:0")
-      AnalyticsHandler.recordAbanonedCartEvent(this.user,this.cart)
+    async triggerAbandonedCartEmail () {
+      if (this.cart && this.cart.items.length > 0 ){
+      const cartItem = await this.getProductByID(this.cart.items[0].product_id)
+      AnalyticsHandler.recordAbanonedCartEvent(this.user,this.cart,cartItem)
+      }
```

Comment:
Why only sending the first product from the cart and not multiple/all products? Guessing it's because of message template complexity? Pinpoint templates support handlebars so `{{each ...}}` may work to iterate of the values in the `ShoppingCartItem...` user attributes.

Reply:
I couldnt find any helpers for iterating over values when referring this [doc](https://docs.aws.amazon.com/pinpoint/latest/userguide/message-template-helpers.html#customhelpers). Even in the exsiting [templates](https://github.com/aws-samples/retail-demo-store/blob/master/workshop/4-Messaging/pinpoint-templates/recommendations-email-template.html), I could see only hardcoded values for loops
Could you provide a link to the documentation?

Reply:
As discussed offline, I am facing some challenges in implementing the iterative approach.
Until we are able to find a solution, lets keep the template as is as it still provides a  personalzed experience.